### PR TITLE
Add docs for volume plugin ls/get

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @thaJeztah

File: docs/extend/plugins_volume.md [hunk 1]
```diff
@@ -164,3 +164,52 @@ this point.
...
+**Response**:
+```
+{
+  "Name": "volume_name",
+  "Mountpoint": "/path/to/directory/on/host",
+  Err: ""
+}
+```
+
+Respond with a string error if an error occurred.
```
Comment (thaJeztah) [inline docs/extend/plugins_volume.md]: Is that a plain string? Or a JSON response with `Err` containing the string?

Does the plugin API use headers to communicate errors (e.g. a 500 internal server error)?

**Comment by thaJeztah** [comment]
Thanks! 1 question, otherwise LGTM

I guess this also needs changes to https://github.com/docker/go-plugins-helpers/tree/master/volume ?

File: docs/extend/plugins_volume.md [hunk 2]
```diff
@@ -164,3 +164,52 @@ this point.
...
+
+Get the volume info.
+
+
+**Response**:
+```
+{
+  "Name": "volume_name",
+  "Mountpoint": "/path/to/directory/on/host",
+  Err: ""
```
Comment (aanand) [inline docs/extend/plugins_volume.md]: I think `Err` needs to be quoted here.

File: docs/extend/plugins_volume.md [hunk 3]
```diff
@@ -164,3 +164,52 @@ this point.
...
+**Response**:
+```
+{
+  Volumes: [
+    {
+      "Name": "volume_name",
+      "Mountpoint": "/path/to/directory/on/host"
+    }
+  ],
+  Err: ""
```
Comment (aanand) [inline docs/extend/plugins_volume.md]: `Volumes` and `Err` should be quoted here too

File: docs/extend/plugins_volume.md (hunk 2 above)
Comment (thaJeztah) [inline docs/extend/plugins_volume.md]: Oh, you're right!

File: docs/extend/plugins_volume.md [hunk 4]
```diff
@@ -164,3 +164,52 @@ this point.
...
+}
```
Comment (thaJeztah) [inline docs/extend/plugins_volume.md]: Here as well; `Volumes` should also be quoted

```JSON
{
	"Volumes": [{
		"Name": "volume_name",
		"Mountpoint": "/path/to/directory/on/host"
	}],
	"Err": ""
}
```

Comment (thaJeztah) [inline docs/extend/plugins_volume.md]: @cpuguy83 perhaps you can change all "fences", and add a JSON hint, i.e.;

```
   ```JSON
```

(also happy to do a follow up PR to fix that in all the Plugin docs at once)

File: docs/extend/plugins_volume.md (hunk 3 above)
Comment (thaJeztah) [inline docs/extend/plugins_volume.md]: oh! sorry for the double post, GitHub didn't show your comment for me :blush:

File: docs/extend/plugins_volume.md (hunk 1 above)
Comment (cpuguy83) [inline docs/extend/plugins_volume.md]: Plain string -- this is the same in the rest of the file, lmk if we should chagne this to be more clear.

Any non-200 response should be used to indicate a protocol error but **not**, for instance, using 404 for a resource that doesn't exist.

**Comment by cpuguy83** [comment]
Updated

**Comment by thaJeztah** [comment]
great! LGTM

ping @moxiegirl for review

File: docs/extend/plugins_volume.md [hunk 5]
```diff
@@ -164,3 +164,52 @@ this point.
...
+
+
+**Response**:
+```json
+{
+  "Name": "volume_name",
+  "Mountpoint": "/path/to/directory/on/host",
+  "Err": ""
+}
+```
```
Comment (srust) [inline docs/extend/plugins_volume.md]: Hi @cpuguy83! This 'Get' response is not correct. The code expects the response to be inside a "Volume" hash, like:

````
{
    "Volume": {
        "Name": "volume_name",
        "Mountpoint": "/path/to/directory/on/host",
    },
    "Err": ""
}
````

in volume/drivers/proxy.go:
````
      req.Name = name
      if err = pp.Call("VolumeDriver.Get", req, &ret); err != nil {                                                                
          return                                                                                                                   
      }                                                                                                                            
  
      volume = ret.Volume    
````

*Possibly* its more consistent to have "Name" and "Mountpoint" at the top-level, like the rest of the API, and have the response in the proxy parsed into a Volume hash.

Easier to change the code or the docs? :-)

Comment (cpuguy83) [inline docs/extend/plugins_volume.md]: Oh you're right. I'll change the doc.
As it is currently coded was intentional.
It is equivalent to a function return following this pattern:
```go
 func Get(name string) (*foo, error)
```

**Comment by moxiegirl** [comment]
lgtm @cpuguy83 thanks!

**Pull request merged**